### DISCUSSION TITLE
React-route-dom v6 how to prevent transitions

### DISCUSSION THREAD

**Discussion by bmarchewka** [remix-run/react-router#8341] [Q&A] [answered]
I used Prompt from react-router-dom v5 to prevent transitions to other pages when, for example, form is dirty. It was used to show confirmation dialog for end users. Prompt component is not available anymore in react-router-dom v6.0.2 (at the moment the latest one). Could you please suggest how to prevent transitions now. Which solution should I used? Are you planning to add this component later to v6?

**Comment by datner** [comment]
Not sure how this went unanswered, this is a common use case.

What I do is create a custom hook to mimic the functionality `const [navigate, allow]  = useMyNavigate(state === "dirty")`
but it's not very scaleable..

**Comment by rosko** [comment]
BTW, `<Prompt>` is forgotten on the [upgrade manual](https://reactrouter.com/docs/en/v6/upgrading/v5). Would be great to have some solution

**Comment by schinniah2020** [comment]
We are facing the same issue. Please give us an instruction of v6 using prompt.

**Comment by rosko** [comment]
I found and adapted for me the following solution: https://gist.github.com/rmorse/426ffcc579922a82749934826fa9f743

It uses some undocumented things of `react-router-dom` but works fine.

**Comment by ryanflorence** [comment, marked as answer]
Bringing prompt back is on the roadmap. https://github.com/orgs/remix-run/projects/5/views/1
